pkg/keycloak/realm/phaseHandler.go

```diff
@@ -237,23 +228,28 @@ func (ph *phaseHandler) reconcileUser(kcUser, specUser *v1alpha1.KeycloakUser, r
 		if specUser.Password != nil {
 			specUser.Password = nil
 		}
-		if !resourcesEqual(kcUser.KeycloakApiUser, specUser.KeycloakApiUser) {
-			specUser.ID = kcUser.ID
-			err := authenticatedClient.UpdateUser(specUser, realmName)
-			if err != nil {
-				return err
+		if !createOnly {
+			if !resourcesEqual(kcUser.KeycloakApiUser, specUser.KeycloakApiUser) {
+				specUser.ID = kcUser.ID
+				err := authenticatedClient.UpdateUser(specUser, realmName)
+				if err != nil {
+					return err
+				}
 			}
+		} else {
+			// Need to set the ID because specUser is used for queries
+			specUser.ID = kcUser.ID
```

Comment:
is this an else to to the `if !createOnly`? if so, let's just put this above it? as it applies in both cases, right?

Reply:
Right, and i think this even applies regardless of the value of `createOnly` as the ID is subsequently required for listing users. Will put it into above the whole if and retest.